### ISSUE TITLE
hvm windows dvd install fail

### ISSUE THREAD

**Issue by wayzard** [QubesOS/qubes-issues#1506]
[guest-winsrvxx-dm.log.txt](https://github.com/QubesOS/qubes-issues/files/58882/guest-winsrvxx-dm.log.txt)

Unable to complete Windows HVM installation via physical DVD nor by *iso*.

When attempting to run installation using external dvd drive (dell esata drive)...
```
qvm-create winsrvxx --hvm --label green
qvm-start winsrvxx --cdrom=/dev/sr0
```
OR
```
qvm-start winsrvxx --cdrom=/dev/disk/by-label/WINDOWS
```
OR
```
qvm-start winsrvxx --cdrom=/home/user/windows.iso
```
at "setup is starting"...
windows produces error:

```
Windows cannot access the installation sources. Verify that the installation sources are accessible, and restart the installation.
```
and then vm stays "transient" until killed...

When re-attempting and clicking clicking "repair your computer" (within Windows guest), produces error:
```
(some text, problem, must restart)
Error code: 80070008
```
and then forces user to restart the vm.

Trying to restart afterwards gives this error (in Qubes):
```
--> Loading the VM (type = HVM)...
--> Starting Qubes DB...
--> Setting Qubes DB info for the VM...
Traceback (most recent call last):
  File "/usr/bin/qvm-start", line 125, in <module>
    main()
  File "/usr/bin/qvm-start", line 109, in main
    xid = vm.start(verbose=options.verbose, preparing_dvm=options.preparing_dvm, start_guid=not options.noguid, notify_function=tray_notify_generic if options.tray else None)
  File "/usr/lib64/python2.7/site-packages/qubes/modules/01QubesHVm.py", line 342, in start
    return super(QubesHVm, self).start(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/qubes/modules/000QubesVm.py", line 1856, in start
    self.create_qubesdb_entries()
  File "/usr/lib64/python2.7/site-packages/qubes/modules/000QubesVm.py", line 1049, in create_qubesdb_entries
    self.qdb.write("/name", self.name)
qubes.qdb.Error: (104, 'Connection reset by peer')
```

Depending on the disk/version of Windows, it will get to a different point in the installation, maybe around a certain percentage of decompression files to install to disk, then fail with the above code.

Host: Qubes 3.0, Xen 4.4.3

**Comment by marmarek** [comment]
On Thu, Dec 10, 2015 at 03:40:41PM -0800, wayzard wrote:
> at "setup is starting"
> windows produces error:
> 
> ```
> Windows cannot access the installation sources Verify that the installation sources are accessible, and restart the installation
> ```
> and then vm stays "transient" until killed

Try increasing memory for that VM. Default 512MB may be too small.

> Trying to restart afterwards gives this error (in Qubes):
> ```
> --> Loading the VM (type = HVM)
> --> Starting Qubes DB
> --> Setting Qubes DB info for the VM
> Traceback (most recent call last):
>   File "/usr/bin/qvm-start", line 125, in <module>
>     main()
>   File "/usr/bin/qvm-start", line 109, in main
>     xid = vmstart(verbose=optionsverbose, preparing_dvm=optionspreparing_dvm, start_guid=not optionsnoguid, notify_function=tray_notify_generic if optionstray else None)
>   File "/usr/lib64/python27/site-packages/qubes/modules/01QubesHVmpy", line 342, in start
>     return super(QubesHVm, self)start(*args, **kwargs)
>   File "/usr/lib64/python27/site-packages/qubes/modules/000QubesVmpy", line 1856, in start
>     selfcreate_qubesdb_entries()
>   File "/usr/lib64/python27/site-packages/qubes/modules/000QubesVmpy", line 1049, in create_qubesdb_entries
>     selfqdbwrite("/name", selfname)
> qubesqdbError: (104, 'Connection reset by peer')
> ```

Do you have updates installed? This should be already fixed. And
generally that bug happened only when you try to restart VM shortly
after its shutdown.

-- 
Best Regards,
Marek Marczykowski-Górecki
Invisible Things Lab
A: Because it messes up the order in which people normally read text.
Q: Why is top-posting such a bad thing?

**Comment by wayzard** [comment]
I have tried to increase memory to 8GB...
```
qvm-create winserverx --hvm --label green --mem 8192
qvm-start winserverx --cdrom=/dev/sr0
```
...and it still gives the error.

I have updated Dom0 with `qvm-update-dom0` and it gives this error when I do the above steps right after a fresh boot.

I have attempted this with Windows Server 2008 and Windows Server 2012. (The error code in Windows Server 2008 may be slightly different. I'll double check that and get back.)
